koku/masu/external/downloader/azure/azure_report_downloader.py

```diff
@@ -118,30 +112,34 @@ def create_daily_archives(
     s3_csv_path = com_utils.get_path_prefix(
         account, Provider.PROVIDER_AZURE, provider_uuid, start_date, Config.CSV_DATA_TYPE
     )
-    time_interval, process_date, date_format = get_processing_date(
+    time_interval, process_date = get_processing_date(
         local_file, s3_csv_path, manifest_id, provider_uuid, start_date, end_date, context, tracing_id
     )
-    with pd.read_csv(local_file, chunksize=settings.PARQUET_PROCESSING_BATCH_SIZE) as reader:
+    with pd.read_csv(
+        local_file, chunksize=settings.PARQUET_PROCESSING_BATCH_SIZE, parse_dates=[time_interval]
+    ) as reader:
         for i, data_frame in enumerate(reader):
             if data_frame.empty:
                 continue
-            intervals = data_frame[time_interval].unique()
-            for interval in intervals:
-                csv_date = datetime.datetime.strptime(interval, date_format)
-                # Adding end here so we dont bother to process future incomplete days (saving plan data)
-                if csv_date >= process_date and csv_date <= end_date:
-                    dates.add(interval)
-            if not dates:
+
+            # Adding end here so we dont bother to process future incomplete days (saving plan data)
+            result_df = data_frame.loc[
+                (data_frame[time_interval] >= process_date) & (data_frame[time_interval] <= end_date)
+            ]
```

Comment:
If we set and sort the index on the time_interval, it potentially makes the next operations easier to write
```suggestion
            data_frame = data_frame.set_index(time_interval).sort_index()

            # Adding end here so we dont bother to process future incomplete days (saving plan data)
            data_frame = data_frame.loc[process_date:end_date]
```

and to save on memory, I would not create the `result_df`. We ought to just overwrite `data_frame`.